### PR TITLE
feat(Wiki.gg): add presence

### LINKED ISSUE #6257: Wiki.gg | wiki.gg
### Website name

wiki.gg

### Website URL

https://wiki.gg/

### Website logo

https://media.discordapp.net/attachments/801353652181925899/977254615642935336/Screenshot_2022-05-20_233025.png

### Prerequisites

- [ ] It is a paid service
- [ ] It displays NSFW content
- [ ] It is region restricted

### Description

This is a wiki, I usually use on playing my games. It has terraria as an offical wiki, whereas fandom is an unofficial wiki

### PR DESCRIPTION
**Pull request by DarkVillager**
## Description 

Resolves #6257
Added the presence.
## Acknowledgements
- [X] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [X] I linted the code by running `yarn format`
- [x] I bumped the version of the presence in accordance with [Semver](https://devhints.io/semver)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

<br>
<details>
<summary> Terraria</summary>

</details>

<details>
<summary> Coromon</summary>

</details>

<details>
<summary> Darkdeity</summary>

</details>

<details>
<summary> Wiki.gg</summary>

</details>

</details>

### CODE REVIEW

File: websites/W/Wiki.gg/presence.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
+const presence = new Presence({
+		clientId: "978186598669758504"
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+let title: string, mTitle: string, search: NodeListOf<HTMLInputElement>;
```
Comment (EncryptedDev) [inline websites/W/Wiki.gg/presence.ts:5]: No

File: websites/W/Wiki.gg/presence.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "logo",
+			startTimestamp: browsingTimestamp
+		},
+		path = document.location.pathname,
```
Comment (EncryptedDev) [inline websites/W/Wiki.gg/presence.ts:12]:
```suggestion
		{ pathname } = document.location,
```

File: websites/W/Wiki.gg/presence.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+					}
+				];
+			} else if (path.includes("wiki") || path.includes("index.php")) {
+				title = `Viewing ${title} Wiki about:`;
+				presenceData.state = document
+					.querySelector("#firstHeading")
+					.textContent.replace(":", " ");
+				presenceData.buttons = [
+					{
+						label: "View Wiki Page",
```
Comment (EncryptedDev) [inline websites/W/Wiki.gg/presence.ts:63]:
```suggestion
						label: "View Page",
```

File: websites/W/Wiki.gg/presence.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+					search[1]?.value
+				)
+					presenceData.state = search[1].value;
+				title = `Searching ${title}'s wiki for:`;
+				presenceData.buttons = [
+					{
+						label: "View Search Result",
+						url: document.location.href
+					}
+				];
```
Comment (EncryptedDev) [inline websites/W/Wiki.gg/presence.ts:55]: You don't need a button here

File: websites/W/Wiki.gg/presence.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+			} else if (
+				path === "/" ||
+				document.querySelector(
+					"#mw-content-text > div.mw-parser-output > div > div:nth-child(1) > div > div:nth-child(1)"
+				)
+			)
+				presenceData.state = "Viewing Homepage";
```
Comment (EncryptedDev) [inline websites/W/Wiki.gg/presence.ts:73]:
```suggestion
				presenceData.state = "Viewing home page";
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/W/Wiki.gg/presence.ts (hunk 5 above)
Comment (DarkVillager) [reply websites/W/Wiki.gg/presence.ts:73]: image

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/W/Wiki.gg/presence.ts (hunk 5 above)
Comment (EncryptedDev) [reply websites/W/Wiki.gg/presence.ts:69]: I'm running of our strings
https://github.com/PreMiD/Localization/blob/46616f565a2a05592e8ae58cab697da9516e06eb/src/Presence/general.json#L162-L164

File: websites/W/Wiki.gg/presence.ts [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+			} else if (
+				pathname === "/" ||
+				document.querySelector(
+					"#mw-content-text > div.mw-parser-output > div > div:nth-child(1) > div > div:nth-child(1)"
+				)
+			)
+				presenceData.state = "Viewing Homepage";
+			presenceData.details = title;
+		}
+	} else presenceData.details = "Browsing...";
```
Comment (EncryptedDev) [inline websites/W/Wiki.gg/presence.ts:72]:
```suggestion
	} else presenceData.details = "Browsing";
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/W/Wiki.gg/dist/metadata.json [hunk 7]
```diff
@@ -0,0 +1,41 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"name": "Dark_Ville",
+		"id": "638080361179512853"
+	},
+	"service": "Wiki.gg",
+	"description": {
+		"en": "Platform operated by creators of gaming experiences. Flexible platform that adjusts to the needs of creators.",
+		"nl": "Platform beheerd door makers van spelervaringen. Flexibel platform dat zich aanpast aan de behoeften van de makers."
```
Comment (EncryptedDev) [inline websites/W/Wiki.gg/dist/metadata.json:10]: Suggestion from @QkeleQ10 
```suggestion
		"nl": "Wij brengen een gemeenschap van ontwikkelaars en editors samen om de best mogelijke ervaring voor gamers te creëren."
```

File: websites/W/Wiki.gg/presence.ts [hunk 8]
```diff
@@ -0,0 +1,76 @@
+const presence = new Presence({
+		clientId: "978186598669758504"
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+let title: string, mTitle: string;
```
Comment (EncryptedDev) [inline websites/W/Wiki.gg/presence.ts:5]: What is `mTitle`?

File: websites/W/Wiki.gg/presence.ts (hunk 5 above)
Comment (EncryptedDev) [reply websites/W/Wiki.gg/presence.ts:69]: Resolved

**Comment by EncryptedDev** [comment]
Heading towards being inactive, please answer the review above

**Comment by ImRodry** [comment]
Rebased to apply the latest changes in the prettier configuration. If you need help with your local branch, please ping me on the #presence-dev channel on Discord.

**Comment by ImRodry** [comment]
Please fix the issues with ESLint

**Review by EncryptedDev — APPROVED** [review]
LGTM

**Review by ImRodry — APPROVED** [review]

**Pull request merged**